qdl.h

```diff
@@ -10,28 +10,50 @@
 
 #define MAPPING_SZ 64
 
-struct libusb_device_handle;
+enum QDL_DEVICE_TYPE
+{
+	QDL_DEVICE_USB,
+};
+
+struct qdl_device
+{
+	enum QDL_DEVICE_TYPE dev_type;
+	int fd;
+
+	int (*open)(struct qdl_device *qdl, const char *serial);
+	int (*read)(struct qdl_device *qdl, void *buf, size_t len, unsigned int timeout);
+	int (*write)(struct qdl_device *qdl, const void *buf, size_t nbytes);
+	void (*close)(struct qdl_device *qdl);
+	void (*set_out_chunk_size)(struct qdl_device *qdl, long size);
 
-struct qdl_device {
-        struct libusb_device_handle *usb_handle;
-        int fd;
+	char *mappings[MAPPING_SZ]; // array index is the id from the device
+};
+
+struct libusb_device_handle;
 
-        int in_ep;
-        int out_ep;
+struct qdl_device_usb
```

Comment:
Can we move this to usb.c too? So that the core is really transport-agnostic?

Reply:
done, however had to reimplement a bit `ramdump.c`, any comments/suggestions are welcome!